tests/unit/viahtml/app_test.py

```diff
@@ -78,7 +79,69 @@ def test_it_configures_logging(self, debug_enabled):
                 level=logging.DEBUG if debug_enabled else logging.INFO,
             )
 
-    @pytest.fixture
-    def Hooks(self):
-        with patch("viahtml.app.Hooks", autospec=True) as Hooks:
-            yield Hooks
+
+class TestApplication:
+    # Test the WSGI middleware wrapping behavior
+    # This is a small amount code which generates a huge and annoying amount of
+    # test code as it's very nested and functional
+
+    STATUS = "200 OK"
+    HEADERS = (("Header", "Value"),)
+
+    def test_it_wraps_pywb(self, app, start_response):
+        print(app.hooks)
+        result = app(sentinel.environ, start_response)
+
+        assert result == start_response.return_value
+
+        start_response.assert_called_with(self.STATUS, Any())
+
+    def test_it_modifies_inbound_headers(self, app, start_response):
+        app(sentinel.environ, start_response)
+
+        modify_inbound = app.hooks.headers.modify_inbound
+        modify_inbound.assert_called_once_with(sentinel.environ)
+        app.app.assert_called_once_with(modify_inbound.return_value, Any.function())
+
+    def test_it_modifies_outbound_headers(self, app, start_response):
+        result = app(sentinel.environ, start_response)
+
+        assert result == start_response.return_value
+        modified_outbound = app.hooks.headers.modify_outbound
+        modified_outbound.assert_called_once_with(self.HEADERS)
+        start_response.assert_called_with(Any(), modified_outbound.return_value)
+
+    @pytest.fixture  # pragma: no cover
+    def start_response(self):
+        return create_autospec(lambda status, headers: None)
+
+    @pytest.fixture  # pragma: no cover
+    def app(self):
+        return Application()
+
+    @pytest.fixture(autouse=True)  # pragma: no cover
+    def FrontEndApp(self):
+        with patch("viahtml.app.FrontEndApp", autospec=True) as FrontEndApp:
+            # It's not in the spec, but this should have this object once
+            # created. It needs to be present, but not anything in particular
+
+            app = FrontEndApp.return_value
+            app.rewriterapp = "something"
+
+            # When the app is called, we need it to fake serving a request
+            app.side_effect = lambda environ, start_request: start_request(
+                self.STATUS, self.HEADERS
+            )
+
+            yield FrontEndApp
+
+    @pytest.fixture(autouse=True)  # pragma: no cover
+    def apply_post_app_hooks(self):
+        with patch("viahtml.app.apply_post_app_hooks") as apply_post_app_hooks:
+            yield apply_post_app_hooks
+
+
+@pytest.fixture(autouse=True)
+def Hooks():
+    with patch("viahtml.app.Hooks", autospec=True) as Hooks:
+        yield Hooks
```

Comment:
I'm guessing that viahtml is missing [the usual `patch` fixture](https://github.com/hypothesis/lms/blob/869911e7093b6cabfca41e9a64828725905f8f38/tests/conftest.py#L62-L74)? Can we add it so that we can patch things the same way as we do in our other apps? For consistency and because it's easier to use and ensures that `autospec` is always used

Reply:
I've added this now. Although I don't greatly like re-adding the same blob of code to every project.

Reply:
Maybe we need a pytest fixtures package or our own pytest plugin that registers a bunch of fixtures or something